Software/lib/GNSE_HAL/GNSE_hal.h

```diff
@@ -0,0 +1,61 @@
+/** Copyright © 2021 The Things Industries B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file GNSE_hal.h
+ *
+ * @copyright Copyright (c) 2021 The Things Industries B.V.
+ *
+ */
+
+#ifndef GNSE_HAL_H
+#define GNSE_HAL_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdbool.h>
+#include "MX25R16.h"
+
+#define GNSE_HAL_INIT_DELAY 100 // TODO: Decrease value to settling time, see https://github.com/TheThingsIndustries/generic-node-se/issues/153
+
+extern MxChip GNSE_HAL_Flash;
```

Comment:
You forgot to add a definition, see this
https://www.geeksforgeeks.org/difference-between-definition-and-declaration/

Please add it in GNSE_hal.c